### PR TITLE
feat: annotation db handler

### PR DESCRIPTION
**Pull request by yadach**
## What?
- Add AnnotationDBHandler

## Why?
- Handle annotation data

### CODE REVIEW

File: pydtk/db/v4/handlers/annotation.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+        password="",
+        database="",
+    )
+    _database_id: str = ""
+
+    def __init__(self, database_id: str = "default", **kwargs):
+        """Initialize AnnotationDBHandler.
+
+        Args:
+            database_id (str): ID of the database.
```
Comment (d-hayashi) [inline pydtk/db/v4/handlers/annotation.py:32]:
```suggestion
            database_id (str): ID of the database.
            
```

File: pydtk/db/v4/handlers/annotation.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        Returns:
+            (DBHandler): Database Handler.
```
Comment (d-hayashi) [inline pydtk/db/v4/handlers/annotation.py:34]:
```suggestion
            (DBHandler): Database Handler.
            
```

File: pydtk/db/v4/handlers/annotation.py [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+        # when grouping annotations by `annotation_id`
+        for column in self.config["columns"]:
+            column["aggregation"] = "last"
+
+    @property
+    def _df_name(self):
+        """Return _df_name.
+
+        Returns:
+            (str): name
```
Comment (d-hayashi) [inline pydtk/db/v4/handlers/annotation.py:94]:
```suggestion
            (str): name
            
```

File: test/test_annotations.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+def test_annotation_handler(
+    db_engine: str,
+    db_host: str,
+    db_username: Optional[str],
+    db_password: Optional[str],
+    db_name: Optional[str],
+    schema_sample_json: str,
+    drop_key: str,
+):
+    """Create DB of records directory.
```
Comment (inuzukam) [inline test/test_annotations.py:39]:
```suggestion
    """Test annotation handler.
```

File: test/test_annotations.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+
+    Args:
+        db_engine (str): DB engine (e.g., 'tinydb')
+        db_host (str): Host of path of DB
+        db_username (str): Username
+        db_password (str): Password
+        db_name (str): Database name
+
+    """
```
Comment (inuzukam) [inline test/test_annotations.py:48]:
```suggestion
    Args:
        db_engine (str): DB engine (e.g., 'tinydb')
        db_host (str): Host of path of DB
        db_username (str): Username
        db_password (str): Password
        db_name (str): Database name
        schema_sample_json: str: Path to schema sample json
        drop_key: str: Key to be be dropped for test
    
    """
```

File: test/test_annotations.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+    # data validation
+    for k in annotation_dict.keys():
+        assert annotation_dict[k] == data[0][k]
+
+    # check schema
+    annotation_dict.pop(drop_key)
+    try:
+        handler.add_data(annotation_dict)
+        raise RuntimeError(f"{drop_key} is dropped.")
```
Comment (inuzukam) [inline test/test_annotations.py:81]:
```suggestion
        raise RuntimeError(f"{drop_key} is dropped but adding data was succeeded unexpectedly.")
```

**Review by inuzukam — CHANGES_REQUESTED** [review]

**Review by inuzukam — APPROVED** [review]

**Comment by inuzukam** [comment]
LGTM！

**Pull request merged**